src/csharp/Microsoft.Spark.Worker/Processor/BroadcastVariableProcessor.cs

```diff
@@ -48,8 +59,10 @@ internal BroadcastVariables Process(Stream stream)
                 {
                     if (broadcastVars.DecryptionServerNeeded)
                     {
-                        throw new NotImplementedException(
-                            "broadcastDecryptionServer is not implemented.");
+                        var readBid = SerDe.ReadInt64(socket.InputStream);
+                        Debug.Assert(bid == readBid);
```

Comment:
Since we get the value from an external system, can we throw an exception instead of assert?

Reply:
Added an exception, please let me know if the wording makes sense, I wasn't completely sure how to frame it.